### PR TITLE
Tests for Controllers

### PR DESCRIPTION
**Pull request by Epaltechs**
### At this point in the project's milestone, I:
- Add RSpec as a gem to my project.
- Create a Request spec file for all my controllers.
- Make sure that for each action:
  - If the response status is correct.
  - If a correct template is rendered.
  - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      get user_posts_path(1)
+      expect(response).to have_http_status(:ok)
+    end
+    it 'renders index template correctly' do
+      get user_posts_path(1)
+      expect(response).to render_template('index')
+    end
+    it 'includes correct placeholder text' do
+      get user_posts_path(1)
+      expect(response.body).to include('Posts for a User')
```
Comment (sja-thedude) [inline spec/requests/posts_controller_spec.rb]: - [ ] Kindly make sure to add tests for GET / SHOW actions for the post controller.

File: spec/requests/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  describe 'GET #index' do
+    before(:example) { get users_path } # get(:index)
+    it 'is a success' do
+      expect(response).to have_http_status(:ok)
+    end
+    it "renders 'index' template" do
+      expect(response).to render_template('index')
+    end
+    it 'includes correct placeholder text' do
+      expect(response.body).to include('List of Users')
```
Comment (sja-thedude) [inline spec/requests/users_controller_spec.rb:13]: - [ ] Kindly make sure to add tests for GET / SHOW actions for the user controller.

**Review by sja-thedude — COMMENTED** [review]
## Required Changes ♻️

Hi @Epaltechs, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Correct Gitflow
✔️ All linters passing

## Required 
- [ ] Kindly make sure to fix this error. When I tried running bundle install it throws this error.
Screenshot 2022-08-30 at 7 42 52 PM

- [ ] Also make sure that all of your test cases pass.

- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Epaltechs** [comment]
#### In this commit, I implement the changes requested by the reviewer, which are:
- Fix the error in the Gemfile.
- Make sure all the test cases pass.
- Add tests for GET / SHOW actions for the post controller.
- Add tests for GET / SHOW actions for the user controller.
image

**Review by arslanbisharat — APPROVED** [review]
Hi @Epaltechs,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**